mage_ai/data_cleaner/cleaning_rules/remove_collinear_columns.py

```diff
@@ -16,27 +16,28 @@ def __init__(self, df, column_types, statistics):
         self.numeric_indices = np.arange(len(self.numeric_df))
 
     def evaluate(self):
+        """ Evaluate columns for collinearity with Variance Inflation Factor.
+        VIF = 1 / (1 - <coefficient of determination on column k>)
+        Measures increase in regression model variance due to multicollinearity
+        => column k is multicollinear with others if model predicting its value
+        has this variance inflation greater than some amount
+        """
         suggestions = []
         if self.numeric_df.empty or len(self.numeric_df) < self.MIN_ENTRIES:
             return suggestions
         collinear_columns = []
         self.numeric_df['intercept'] = np.ones(len(self.numeric_df))
-        for column in self.numeric_columns[:-1]:
-            variance_inflation_factor = self.get_variance_inflation_factor(column)
-            if variance_inflation_factor > self.VIF_UB:
-                collinear_columns.append(column)
-                self.numeric_df.drop(column, axis=1, inplace=True)
-        if len(collinear_columns) != len(self.numeric_columns) - 1:
-            # check the final column if and only if there are other columns to compare it to
-            column = self.numeric_columns[-1]
-            variance_inflation_factor = self.get_variance_inflation_factor(column)
-            if variance_inflation_factor > self.VIF_UB:
-                collinear_columns.append(column)
+
+        C = np.corrcoef(df.to_numpy().T)
+        vifs = np.diagonal(np.linalg.pinv(C))
+        collinear_columns = [num_col for vif, num_col in
+                                 zip(vifs, self.numeric_columns) if vif > self.VIF_UB]
         if len(collinear_columns) != 0:
             suggestions.append(
                 self._build_transformer_action_suggestion(
                     'Remove collinear columns',
-                    'Delete these columns to remove redundant data and increase data quality.',
+                    'Delete some of these columns to remove'
```

Comment:
There should be a single whitespace at the end of this line for the space between "remove" and "redundant". Currently this gets printed out as "...to removeredundant data and..."